x/core/keeper/gas_meter.go

```diff
@@ -63,9 +62,82 @@ func (k Keeper) DistributionsFromGasMeter(ctx sdk.Context, reqID string, reqHeig
 	return dists
 }
 
+func (k Keeper) ProcessDistributions(ctx sdk.Context, dists []types.Distribution, dr *types.DataRequest, minimumStake math.Int) error {
+	// Distribute in order.
+	denom, err := k.stakingKeeper.BondDenom(ctx)
+	if err != nil {
+		return err
+	}
+
+	remainingEscrow := dr.Escrow
+
+	// TODO Events
+	var amount math.Int
+	for _, dist := range dists {
+		switch {
+		case dist.Burn != nil:
+			amount = math.MinInt(dist.Burn.Amount, remainingEscrow)
+			err = k.bankKeeper.BurnCoins(ctx, types.ModuleName, sdk.NewCoins(sdk.NewCoin(denom, amount)))
+			if err != nil {
+				return err
+			}
+
+		case dist.DataProxyReward != nil:
+			amount = math.MinInt(dist.DataProxyReward.Amount, remainingEscrow)
+			payoutAddr, err := sdk.AccAddressFromBech32(dist.DataProxyReward.PayoutAddress)
+			if err != nil {
+				// Should not be reachable because the address has been validated.
```

Comment:
I know we aren't doing this in this PR. But maybe a log that says unreachable and reason. This way if it ever pops up in the logs we can appropiately panic lol

Reply:
It is logged in Tally()